components/movementsensor/gpsrtkpmtk/gpsrtkpmtk.go

```diff
@@ -682,3 +698,80 @@ func (g *rtkI2C) Close(ctx context.Context) error {
 
 	return nil
 }
+
+// sendGGAMessage sends GGA messages to the serial port. This is only used to get NTRIP stream
+// from a virtual reference point.
+func (g *rtkI2C) sendGGAMessage() {
+	// Open an I2C handle to read NMEA messages
+	handle, err := g.bus.OpenHandle(g.addr)
+	if err != nil {
+		g.logger.Errorf("can't open gps i2c handle: %s", err)
+		g.err.Set(err)
+		return
+	}
+	defer func() {
+		if closeErr := handle.Close(); closeErr != nil {
+			g.logger.Errorf("error while closing I2C handle after reading GGA message: %v", closeErr)
+			g.err.Set(closeErr)
+		}
+	}()
+
+	// Create a buffer to collect NMEA messages
+	messageBuffer := make([]byte, 0, 1024)
+
+	for !g.ntripStatus {
+		select {
+		case <-g.cancelCtx.Done():
+			return
+		default:
+		}
+
+		// Read from the I2C device
+		buffer, err := handle.Read(g.cancelCtx, 1024)
+		if err != nil {
+			g.err.Set(err)
+			return
+		}
+
+		// Process the received data
+		for _, b := range buffer {
+			// PMTK uses CRLF line endings to terminate sentences, but just LF to blank data.
+			// Since CR should never appear except at the end of our sentence, we use that to determine sentence end.
+			// LF is merely ignored.
+			if b == 0x0D {
+				if len(messageBuffer) > 0 {
+					// Convert the received bytes to a string
+					receivedData := string(messageBuffer)
+
+					// Check for NMEA GGA messages
+					messages := strings.Split(receivedData, "\n")
```

Comment:
I thought line 760 was already filtering out newlines. I wonder if we can get rid of that and keep this one instead, because this one is easier to understand.